src/Medidata.ZipkinTracer.Core/TraceProvider.cs

```diff
@@ -72,11 +73,11 @@ internal TraceProvider(IZipkinConfig config, IOwinContext context = null)
                 headerSampled = context.Request.Headers[SampledHeaderName];
             }
 
-            TraceId = Parse(headerTraceId) ? headerTraceId : GenerateHexEncodedInt64FromNewGuid();
-            SpanId = Parse(headerSpanId) ? headerSpanId : TraceId;
-            ParentSpanId = Parse(headerParentSpanId) ? headerParentSpanId : string.Empty;
+            TraceId = headerTraceId.IsParsableTo128Or64Bit() ? headerTraceId : GenerateHexFromNewGuid(config.Create128BitTraceId);
```

Comment:
> PS you can go ahead and test this, too. just start zipkin and send bad
trace ids to its endpoint. Nothing will show up, so you will have no means
to debug the diagram mentioned.

Will try adhoc-ing this with @bvillanueva-mdsol.

Seems like Zipkin is not logging or returning `BadRequest` when it receives malformed data.
In that case re-starting the trace is a solution.
This means by swallowing the malformed id we would never know that we have a bug somewhere and we would have requests with broken traces.
If everyone is okay with this I'm okay as well.